cachi2/core/package_managers/pip.py

```diff
@@ -237,6 +257,49 @@ def get_version(self) -> Optional[str]:
         """Attempt to determine the package version. Should only be called if file exists."""
 
 
+class PyProjectTOML(SetupFile):
+    """Extract project name and version from a pyproject.toml file."""
+
+    def __init__(self, top_dir: RootedPath) -> None:
+        """
+        Initialize a PyProjectTOML.
+
+        :param top_dir: Path to root of project directory
+        """
+        super().__init__(top_dir, "pyproject.toml")
+
+    def get_name(self) -> Optional[str]:
+        """Get project name if present."""
+        try:
+            return self._parsed_toml["project"]["name"]
+        except KeyError:
+            log.info("Invalid configuration 'name' is a required property")
```

Comment:
For consistency with SetupCFG, could this be `No project.name in pyproject.toml`?

I would avoid saying "Invalid" in the log message, since the `project` property is not required in pyproject.toml